pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/NullValueTransformer.java

```diff
@@ -52,6 +64,12 @@ public GenericRow transform(GenericRow record) {
         record.putDefaultNullValue(fieldName, entry.getValue());
       }
     }
+
+    // handle null value in time column
+    if (_timeConverter != null && record.getValue(_timeColumnName) == null) {
+      record.putDefaultNullValue(_timeColumnName, _timeConverter.fromMillisSinceEpoch(System.currentTimeMillis()));
```

Comment:
The value should be converted into the correct data type. One way to achieve that is to put the `NullValueTransformer` after the `DataTypeTransformer` in the `CompositeTransformer`

Reply:
+1. plz add a test for different time granularity